### DISCUSSION TITLE
LaTeX Workshop Intellisense not indexing all commands

### DISCUSSION THREAD

**Discussion by janwil123** [James-Yu/LaTeX-Workshop#4047] [Q&A]
I am using VSCode + LaTeX-Workshop for a math book that makes heavy use of tikz-euclide package. It is not one of the 'standard' packages in the sense that it does not come with its own .cwl file. This should be OK as the [LaTeX-Workshop Intellisense documentation](https://github.com/James-Yu/LaTeX-Workshop/wiki/Intellisense) states: "The files of a LaTeX project are searched for any already used commands in the form mycommand followed by several {} groups. Then, a snippet is dynamically built for each of them and they are added to the command completion list."

I seem to remember that up to some point the tikz-euclide commands I used in the project indeed appeared as snippets, but not any more. This is sad because the tikz-euclide commands are rather long. It is still possible that I myself am doing something wrong, but I am unsure how to debug this problem. Where can I find the current command completion list for example? How can I check which files are the ones that intellisense engine actually scans? Do I need to have some config options that I should re-check?

Any help would be greatly appreciated!
Jan

**Comment by jlelong** [comment]
Unfortunately `tikz` related code is too different from standard LaTeX code to be parsed safely by the javascript parser we use. Hence, we skip the `tikzpicture` environment when looking for already used commands

https://github.com/James-Yu/LaTeX-Workshop/blob/edb8b1792c33bb1ca7d9c406de79c136dc9b8509/src/providers/completer/reference.ts#L171-L173

We should update the documentation accordingly.

Your best option is to write a limited `cwl` file for commands you use from `tikz-euclide`. Then, follow the steps described in https://github.com/James-Yu/LaTeX-Workshop/wiki/Intellisense#commands-starting-with- to integrate your own `cwl` file into LW.
